core/src/main/java/org/mskcc/cbio/oncokb/bo/impl/EvidenceBoImpl.java

```diff
@@ -173,34 +279,78 @@
+            if (CacheUtils.isEnabled()) {
+                for (Evidence evidence : CacheUtils.getEvidences(gene)) {
+                    if (evidenceTypes.contains(evidence.getEvidenceType())) {
+                        for (TumorType oncoTreeType : tumorTypes) {
+                            if (oncoTreeType.getCode() != null) {
+                                if (oncoTreeType.getCode().equals(evidence.getSubtype())) {
+                                    set.add(evidence);
+                                }
+                                if (oncoTreeType.getMainType() != null && oncoTreeType.getMainType().getName().equals(evidence.getCancerType()) && evidence.getSubtype() == null) {
+                                    set.add(evidence);
+                                }
+                            } else if (oncoTreeType.getMainType() != null) {
+                                if (oncoTreeType.getMainType().getName().equals(evidence.getCancerType()) && evidence.getSubtype() == null) {
+                                    set.add(evidence);
+                                }
+                            }
+                        }
+                    }
+                }
+            } else {
+                for (EvidenceType evidenceType : evidenceTypes) {
+                    for (TumorType oncoTreeType : tumorTypes) {
+                        if (oncoTreeType.getCode() != null) {
+                            set.addAll(getDao().findEvidencesByGeneAndSubtype(gene, evidenceType, oncoTreeType.getCode()));
+                            if (oncoTreeType.getMainType() != null) {
+                                set.addAll(getDao().findEvidencesByGeneAndCancerTypeNoSubtype(gene, evidenceType, oncoTreeType.getMainType().getName()));
+                            }
+                        } else if (oncoTreeType.getMainType() != null) {
+                            set.addAll(getDao().findEvidencesByGeneAndCancerType(gene, evidenceType, oncoTreeType.getMainType().getName()));
                         }
-                    } else if (oncoTreeType.getMainType() != null) {
-                        set.addAll(getDao().findEvidencesByGeneAndCancerType(gene, evidenceType, oncoTreeType.getMainType().getName()));
                     }
                 }
             }
         }
-        return new ArrayList<Evidence>(set);
+        return new ArrayList<>(set);
     }
 
     @Override
     public List<Evidence> findEvidencesByIds(List<Integer> ids) {
-        return getDao().findEvidencesByIds(ids);
+        if (CacheUtils.isEnabled()) {
+            List<Evidence> evidences = new ArrayList<>();
+            for (Evidence evidence : CacheUtils.getAllEvidences()) {
+                if (ids.contains(evidence.getId())) {
+                    evidences.add(evidence);
+                }
+                if (evidences.size() == ids.size()) {
```

Comment:
is it possible?
And useless to put it here, because the condition won't be met until the for loop has iterated each item